<?xml version="1.0" encoding="utf-8"?>
<launch>
	<arg name="robot" default="X1" />
	<arg name="base_frame" default="X1/base_footprint"/>
    <arg name="odom_frame" default="X1/odom"/>
    <arg name="map_frame" default="slam_map"/>
    <arg name="scan" default="/X1/scan"/>

	<include file="$(find pointcloud_to_laserscan)/launch/pcl2laser.launch">
		<arg name="robot" value="$(arg robot)" />
	</include>

	<include file="$(find mapping)/launch/pub_tf.launch"></include>

	<include file="$(find gmapping)/launch/slam_gmapping_pr2.launch">
		<arg name="base_frame" value="$(arg base_frame)"/>
		<arg name="map_frame" value="$(arg map_frame)"/>
		<arg name="odom_frame" value="$(arg odom_frame)"/>
		<arg name="scan" value="$(arg scan)"/>
	</include>
</launch>